test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -223,78 +237,295 @@ def test_post_process(setup_and_teardown):
 
     wait_to_upload_assemblies = {
         assembly_name: assembly_dir
         for assembly_name, assembly_dir in zip(ASSEMBLY_NAMES, assembly_dirs)
     }
 
-    conf = Mock()
-    conf.workers = 5
-    conf.input_queue = Queue()
-    conf.output_queue = Queue()
-
-    # an uploaded successful
-    conf.output_queue.put((ASSEMBLY_NAMES[0], "12345_58_1"))
-    # an upload failed
-    conf.output_queue.put((ASSEMBLY_NAMES[1], None))
-
-    failed_names = workspace_uploader._upload_assembly_files_in_parallel(
-        conf, "CI", 12345, upload_dir, wait_to_upload_assemblies
-    )
+    # ws.get_object_info3() is unused in this test case
+    ws = create_autospec(Workspace, spec_set=True, instance=True)
+    asu = create_autospec(AssemblyUtil, spec_set=True, instance=True)
+    asu.save_assemblies_from_fastas.return_value = {
+        "results": [
+            {"upa": "12345/58/1"},
+            {"upa": "12345/60/1"}
+        ]
+    }
 
-    expected_tuple1 = (
+    uploaded_count = workspace_uploader._upload_assembly_files_in_parallel(
+        asu,
+        ws,
         "CI",
         12345,
-        os.path.join(
-            workspace_uploader.JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER, ASSEMBLY_NAMES[0]
-        ),
-        assembly_dirs[0],
-        ASSEMBLY_NAMES[0],
+        "214",
         upload_dir,
-        1,
-        len(ASSEMBLY_NAMES),
+        wait_to_upload_assemblies,
+        2,
+        output_dir,
+    )
+
+    assert uploaded_count == 2
+
+    # assert that no interactions occurred with ws
+    ws.get_object_info3.assert_not_called()
+
+    # assert that asu was called correctly
+    asu.save_assemblies_from_fastas.assert_called_once_with(
+        {
+            "workspace_id": 12345,
+            "inputs": [
+                {
+                    "file": "/kb/module/work/tmp/GCF_000979855.1_gtlEnvA5udCFS_genomic.fna.gz",
+                    "assembly_name": "GCF_000979855.1_gtlEnvA5udCFS_genomic.fna.gz",
+                    "object_metadata": {"load_id": "214"},
+                },
+                {
+                    "file": "/kb/module/work/tmp/GCF_000979175.1_gtlEnvA5udCFS_genomic.fna.gz",
+                    "assembly_name": "GCF_000979175.1_gtlEnvA5udCFS_genomic.fna.gz",
+                    "object_metadata": {"load_id": "214"},
+                }
+            ]
+        }
     )
 
-    expected_tuple2 = (
+    # check softlink for post_process
+    assert os.readlink(os.path.join(upload_dir, "12345_58_1")) == os.path.join(
+        output_dir, "12345_58_1"
+    )
+    assert os.readlink(os.path.join(upload_dir, "12345_60_1")) == os.path.join(
+        output_dir, "12345_60_1"
+    )
+
+    # check hardlink for post_process
+    assert os.path.samefile(
+        src_files[0],
+        os.path.join(os.path.join(output_dir, "12345_58_1"), "12345_58_1.fna.gz")
+    )
+
+    assert os.path.samefile(
+        src_files[1],
+        os.path.join(os.path.join(output_dir, "12345_60_1"), "12345_60_1.fna.gz")
```

Comment:
```suggestion
        Path(output_dir) /  "12345_58_1" / "12345_58_1.fna.gz"
    )

    assert os.path.samefile(
        src_files[1],
        Path(output_dir) / 12345_60_1 / "12345_60_1.fna.gz"
```

Reply:
yeah, cannot believe I used `os.path.join` twice.

Reply:
I think o.p.join() may take multiple arguments, so that's another option

Reply:
Oh, that's what you did :P

Reply:
:+1: